compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -1376,7 +1376,11 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(using Context) {
       super.typedValDef(vdef1, sym)
 
     override def typedApply(tree: untpd.Apply, pt: Type)(using Context): Tree =
-      val res = constToLiteral(betaReduce(super.typedApply(tree, pt))) match {
+      def cancelQuotes(tree: Tree): Tree =
+        tree match
+          case Quoted(Spliced(inner)) => inner
+          case _ => tree
+      val res = cancelQuotes(constToLiteral(betaReduce(super.typedApply(tree, pt)))) match {
```

Comment:
The fix failed the following code:

```Scala
inline def test[T](inline t: T): T = ${ identity('{ identity(${ identity('{ identity(${ identity('t) }) }) }) }) }
```

Reply:
That is a different problem. I will open an issue for it. See #12225.

Reply:
What about this one:

```Scala
inline def test[T](inline t: T): T = ${ '{ ${ '{ ${ 't } } } } }
```

Reply:
That one works. Added it as a regression test.